### ISSUE TITLE
Move ZTNCUI 

### ISSUE THREAD

**Issue by jimthedj65** [key-networks/ztncui#66]
Hi All,

I am looking to move my controller from one machine to another. what directories would I need to move to the new machine to continue.

Currently using latest version of ztncui with recent version of zerotier.

thanks for any guidance.

**Comment by lideming** [comment]
Hi, I have done this before.

Assuming you run ZT and ztncui directly on the OS (not in the Docker):

1) Stop ZT controller and ztncui.
2) Move data of ztncui (`/opt/key-networks/ztncui/etc`) and ZT controler (`/var/lib/zerotier-one`).
3) On the new machine, start ZT controller and ztncui.
4) Ensure they run well on the new machine.
5) On the old machine, ensure the ZT controller won't automatically start again.
  (Or, delete `/var/lib/zerotier-one` and reinstall ZT to make it have a new identity.)

**Comment by jimthedj65** [comment]
I think a merge function would be really helpful for merging two controllers.....

**Comment by MANKUD** [comment]
@lideming I have a question if the new zero tier controller has a different public IP comparing to the old will it still work without issues after moving?

**Comment by lideming** [comment]
> @lideming I have a question if the new zero tier controller has a different public IP comparing to the old will it still work without issues after moving?

Yes, it will work.

Controllers are not binding to IPs. They are just normal peers in ZeroTier VL1 perspective, so they can have changing public IP or even hehind NAT. Peers will find each other over ZeroTier roots (and your moons, if available).

**Comment by MANKUD** [comment]
@lideming thank you so much for the quick reply really appreciate it.

**Comment by MANKUD** [comment]
@lideming On the new machine I need to install zero tier and ztncui and then replace with the files copied from the old machine right?

**Comment by lideming** [comment]
@MANKUD Yes, and check your network is working before deleting data from the old machine.

Also ensure the old machine will never run the old ZeroTier controller and using the old id, or they will conflict with each other.

**Comment by MANKUD** [comment]
@lideming Noted I will try it and if any issues will report it here.

**Comment by MANKUD** [comment]
@lideming getting this error on the new machine followed the steps you mentioned exactly.

![image](https://user-images.githubusercontent.com/83474708/181595477-7b7671f4-aa10-4d91-ac4d-8ce9e45b99f0.png)

**Comment by MANKUD** [comment]
> @lideming getting this error on the new machine followed the steps you mentioned exactly.
> 
> ![image](https://user-images.githubusercontent.com/83474708/181595477-7b7671f4-aa10-4d91-ac4d-8ce9e45b99f0.png)

Server reboot seems to have fixed the issue. I am able to establish connections.

**Comment by MANKUD** [comment]
@lideming I am having a weird issue where I am trying to rename a member from the member name field it just logs me out and gets me back to the login page.. Very strange I did not have such issue on the old machine rest everything seems fine. Do you have any idea why it's happening and how to fix this as assigning names to the member is very important for me.

**Comment by MANKUD** [comment]
> @lideming I am having a weird issue when I am trying to rename a member from the member name field it just logs me out and gets me back to the login page.. Very strange I did not have such issue on the old machine rest everything seems fine. Do you have any idea why it's happening and how to fix this as assigning names to the member is very important for me.

 I fixed it by changing permissions.

**Comment by rupesh43210** [comment]
> Hi, I have done this before.
> 
> Assuming you run ZT and ztncui directly on the OS (not in the Docker):
> 
> 1. Stop ZT controller and ztncui.
> 2. ~Move~ Copy data of ztncui (`/opt/key-networks/ztncui/etc`) and ZT controler (`/var/lib/zerotier-one`).
> 3. On the new machine, start ZT controller and ztncui.
> 4. Ensure they run well on the new machine.
> 5. On the old machine, ensure the ZT controller won't automatically start again.
>    (Or, delete `/var/lib/zerotier-one` and reinstall ZT to make it have a new identity.)

I am unable to restart ztncui after I copy all the directories mentioned. ( I am replacing both directories mentioned) is that how it is to be done or I am doing something wrong here? 

Kindly help

![image](https://user-images.githubusercontent.com/85641027/219936594-043047f0-b0b2-4d4d-949c-75a76f68570a.png)
